xen/arch/arm/coproc/coproc.h

```diff
@@ -29,28 +29,124 @@
 
 #include "schedule.h"
 
-/* coproc memory range */
-struct mmio {
+/*
+ * In fields and function names following prefixes mean:
+ * - [p] Platform specific. Means coprocessor IP specific stuff.
+ * - [m] Machine specific. Means some stuff specific to a SoC which integrates
+ *       a coprocessor.
+ * - [v] Virtual. Means a virtual coprocessor related stuff.
+ */
+
+/*
+ * Coprocessor platform memory mapped IO range description
+ *
+ * This is a structure what describes one particular mmio range of a
+ * coprocessor IP block.
+ * This description expected to be common for different SoCs which could
+ * integrate this particular coprocessor IP.
+ * This structure is used as static with following fields initialized at the
+ * compilation time:
+ *     name - to find in DT, ignored if a coproc has only one mmio range
+ *     size - >0 for sanity check, =0 - no sanity check needed, filled from DT
+ *               (f.e. for SRAM bank)
+ *     ops  - IO registers access emulation (r/w) handlers
+ */
+struct pcoproc_mmio {
+    char *name;
+    u64 size;
```

Comment:
no, u64, or paddr_t, as a value comparable to one received by dt_device_get_address() from a device tree.